File: janusgraph-dotnet/src/JanusGraph.Net.Extensions/IO/GraphSON/PointSerializer.cs
#region License

/*
 * Copyright 2018 JanusGraph Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#endregion

using System.Collections.Generic;
using Gremlin.Net.Structure.IO.GraphSON;
using JanusGraph.Net.Extensions.Geoshapes;

namespace JanusGraph.Net.Extensions.IO.GraphSON
{
    internal class PointSerializer : IGraphSONSerializer
    {
        public Dictionary<string, dynamic> Dictify(dynamic objectData, GraphSONWriter writer)
        {
            var point = (Point) objectData;
            return GraphSONUtil.ToTypedValue("Geoshape",
                new Dictionary<string, object> {{"coordinates", new[] {point.Longitude, point.Latitude}}},
                "janusgraph");
        }
    }
}

File: janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/io/binary/geoshape/PointSerializer.java
// Copyright 2021 JanusGraph Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package org.janusgraph.graphdb.tinkerpop.io.binary.geoshape;

import org.apache.tinkerpop.gremlin.structure.io.Buffer;
import org.apache.tinkerpop.gremlin.structure.io.binary.GraphBinaryReader;
import org.apache.tinkerpop.gremlin.structure.io.binary.GraphBinaryWriter;
import org.janusgraph.core.attribute.Geoshape;
import org.janusgraph.graphdb.tinkerpop.io.binary.GeoshapeGraphBinaryConstants;

public class PointSerializer extends GeoshapeTypeSerializer {

    public PointSerializer() { super(GeoshapeGraphBinaryConstants.GEOSHAPE_POINT_TYPE_CODE); }

    @Override
    public Geoshape readNonNullableGeoshapeValue(final Buffer buffer, final GraphBinaryReader context) {
        final double latitude = buffer.readDouble();
        final double longitude = buffer.readDouble();
        return Geoshape.point(latitude, longitude);
    }

    @Override
    public void writeNonNullableGeoshapeValue(final Geoshape geoshape, final Buffer buffer, final GraphBinaryWriter context) {
        final Geoshape.Point point = geoshape.getPoint();
        buffer.writeDouble(point.getLatitude());
        buffer.writeDouble(point.getLongitude());
    }
}
